UglyTicTacToe/UglyTicTacToe/Computer.aspx.cs

```diff
@@ -0,0 +1,340 @@
+                        case 1:
+                            ab.Text = "O";
+                            ab.Enabled = false;
+                            break;
+                        case 2:
+                            ac.Text = "O";
+                            ac.Enabled = false;
+                            break;
+                    }
+                    break;
+                case 1:
+                    switch (col)
+                    {
+                        case 0:
+                            ba.Text = "O";
+                            ba.Enabled = false;
+                            break;
+                        case 1:
+                            bb.Text = "O";
+                            bb.Enabled = false;
+                            break;
+                        case 2:
+                            bc.Text = "O";
+                            bc.Enabled = false;
+                            break;
+                    }
+                    break;
+                case 2:
+                    switch (col)
+                    {
+                        case 0:
+                            ca.Text = "O";
+                            ca.Enabled = false;
+                            break;
+                        case 1:
+                            cb.Text = "O";
+                            cb.Enabled = false;
+                            break;
+                        case 2:
+                            cc.Text = "O";
+                            cc.Enabled = false;
+                            break;
+                    }
+                    break;
+            }
+        }
+
+        protected void aa_Click(object sender, EventArgs e)
+        {
+            TakeTurn(aa, 0, 0);
+        }
+
+        protected void ab_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ab, 0, 1);
+        }
+
+        protected void ac_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ac, 0, 2);
+        }
+
+        protected void ba_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ba, 1, 0);
+        }
+
+        protected void bb_Click(object sender, EventArgs e)
+        {
+            TakeTurn(bb, 1, 1);
+        }
+
+        protected void bc_Click(object sender, EventArgs e)
+        {
+            TakeTurn(bc, 1, 2);
+        }
+
+        protected void ca_Click(object sender, EventArgs e)
+        {
+            TakeTurn(ca, 2, 0);
+        }
+
+        protected void cb_Click(object sender, EventArgs e)
+        {
+            TakeTurn(cb, 2, 1);
+        }
+
+        protected void cc_Click(object sender, EventArgs e)
+        {
+            TakeTurn(cc, 2, 2);
+        }
+
+        protected bool Win(Player player)
+        {
+            if (Board.GetTurnNumber() > 9)
+            {
+                Winner.Text = "You have Tied";
+            }
+            //check rows
+            for (int row = 0; row < Board.GetBoard().GetLength(0); row++)
```

Comment:
Again, I'd make just a single call to the `Board.GetBoard` method, storing the result in a local `board` variable. Or you could consider putting this logic into a `Board.CheckWin` method.